hack/create-dev-cluster.sh

```diff
@@ -23,6 +23,9 @@ set -o pipefail
 : "${AZURE_CLIENT_ID:?Environment variable empty or not defined.}"
 : "${AZURE_CLIENT_SECRET:?Environment variable empty or not defined.}"
 
+ENVSUBST="${PWD}/hack/tools/bin/envsubst-v1.1.0"
```

Comment:
you should be able to use the symlink `/hack/tools/bin/envsubst` instead of having to pin to a version, and then instead of running make ${ENVSUBST} below just add `$(ENVSUBST)` as a dependency to a new makefile target called at the beginning of this script.

This is to avoid having the version hardcoded somewhere else than the makefile (which would require updating it whenever we bump the version).